jekyll/_cci2/caching.md

```diff
@@ -189,6 +189,22 @@ neither `save_cache` nor `restore_cache` support globbing for the `paths` key.
 
 ## Using Keys and Templates
 
+A cache-key is a _user-defined_ string that corresponds to a data cache. A cache-key can be created by interpolating **dynamic values** — these are called **templates**. Anything you see in a cache-key between curly braces is a template. Consider the following example:
+
+```sh
+{% raw %}myapp-{{ checksum "package-lock.json" }}{% endraw %}
+```
+
+The above example will output a unique string to represent this key. Here, the example is using a [checksum](https://en.wikipedia.org/wiki/Checksum) to create a unique string that represents the contents of a `package-lock.json` file.
+
+The example may output a string that looks like the following:
+
+```sh
+{% raw %}myapp-dlkj3oadlkfgjadflkje3ijaoi43oi49odlkja$lakdjfg#lakdj4Djlae<etc>{% endraw %}
```

Comment:
[Caching Dependencies - CircleCI](https://circleci.com/docs/2.0/caching/) says:
> A base64 encoded SHA256 hash of the given filename’s contents, so that a new cache key is generated if the file changes.

So I think the characters such as `$` and `#` should not appear in the result of the checksum.
In my job result page, I can see the string something like this: `myapp-+V9MbGBceJh_zOWQI6ga3gDPGB26mbhWJxUYmiEidlQ=`

Reply:
Ha! Good point @shuuji3 - I was going for "just imagine a garbled-string to be the checksum" — I should have used something more realistic to a checksum. I'll make an update and remove the bogus chars.